### PR TITLE
[Merged by Bors] - feat: more NNRat lemmas

### PR DESCRIPTION
**Pull request by eric-wieser**
Note that the previous version of `Rat.ofScientific_eq_ofScientific` was not well-formed, as it did not include the necessary `ofNat`s.

---

Some steps towards

```lean
open OfNat in
theorem _root_.ofScientific_def {K} [DivisionSemiring K] (m : ℕ) (s : Bool) (e : ℕ) :
    (OfScientific.ofScientific m s e : K)
      = (ofNat m : ℕ) * (10 : K) ^ (cond s (-(ofNat e)) (ofNat e) : ℤ) := by
```

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Field/Defs.lean [hunk 1]
```diff
@@ -229,5 +229,5 @@ end Rat
 
 /-- `OfScientific.ofScientific` is the simp-normal form. -/
 @[simp]
-theorem Rat.ofScientific_eq_ofScientific :
-    Rat.ofScientific = OfScientific.ofScientific := rfl
+theorem Rat.ofScientific_eq_ofScientific (m : ℕ) (s : Bool) (e : ℕ):
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Field/Defs.lean:232]:
```suggestion
theorem Rat.ofScientific_eq_ofScientific (m : ℕ) (s : Bool) (e : ℕ) :
```

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Data/NNRat/Defs.lean [hunk 2]
```diff
@@ -365,10 +367,14 @@ theorem natAbs_num_coe : (q : ℚ).num.natAbs = q.num := rfl
 
 lemma coprime_num_den (q : ℚ≥0) : q.num.Coprime q.den := by simpa [num, den] using Rat.reduced _
 
--- TODO: Rename `Rat.coe_nat_num`, `Rat.intCast_den`, `Rat.ofNat_num`, `Rat.ofNat_den`
```
Comment (YaelDillies) [inline Mathlib/Data/NNRat/Defs.lean:368]: Sorry, why are you removing this TODO? Here's the correct updated version
```
-- TODO: Rename `intCast_num`, `intCast_den`, `ofNat_num`, `ofNat_den`, `inv_natCast_num`,
-- `inv_natCast_den`, `inv_intCast_num`, `inv_intCast_den`
```

File: Mathlib/Data/Rat/Field.lean [hunk 3]
```diff
@@ -69,4 +69,13 @@ lemma inv_def (q : ℚ≥0) : q⁻¹ = divNat q.den q.num := by ext; simp [Rat.i
 lemma div_def (p q : ℚ≥0) : p / q = divNat (p.num * q.den) (p.den * q.num) := by
   ext; simp [Rat.div_def', num_coe, den_coe]
 
+lemma num_inv_of_ne_zero {q : ℚ≥0} (hq : q ≠ 0) : (q⁻¹).num = q.den := by
```
Comment (YaelDillies) [inline Mathlib/Data/Rat/Field.lean:72]:
```suggestion
lemma num_inv_of_ne_zero {q : ℚ≥0} (hq : q ≠ 0) : q⁻¹.num = q.den := by
```

File: Mathlib/Data/Rat/Field.lean [hunk 4]
```diff
@@ -69,4 +69,13 @@ lemma inv_def (q : ℚ≥0) : q⁻¹ = divNat q.den q.num := by ext; simp [Rat.i
...
+  rw [inv_def, divNat, num, coe_mk, Rat.divInt_ofNat, ← Rat.mk_eq_mkRat _ _ (num_ne_zero.mpr hq),
+    Int.natAbs_ofNat]
+  simpa using q.coprime_num_den.symm
+
+lemma den_inv_of_ne_zero {q : ℚ≥0} (hq : q ≠ 0) : (q⁻¹).den = q.num := by
```
Comment (YaelDillies) [inline Mathlib/Data/Rat/Field.lean:77]:
```suggestion
lemma den_inv_of_ne_zero {q : ℚ≥0} (hq : q ≠ 0) : q⁻¹.den = q.num := by
```

**Review by YaelDillies — COMMENTED** [review]
Sorry I didn't see earlier!

File: Mathlib/Data/NNRat/Defs.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/Data/NNRat/Defs.lean:368]: Surely `ofNat_num` and the `intCast` ones shouldn't be in that list as they doesn't exist?

Comment (YaelDillies) [reply Mathlib/Data/NNRat/Defs.lean:368]: https://leanprover-community.github.io/mathlib4_docs/Std/Data/Rat/Lemmas.html#Rat.ofNat_num ?

Comment (YaelDillies) [reply Mathlib/Data/NNRat/Defs.lean:368]: Oh I see the TODO is in the wrong file. Can you move it `Data.Rat.Defs`

Comment (eric-wieser) [reply Mathlib/Data/NNRat/Defs.lean:368]: I've just put the TODO back; if you want to clarify it further, please do so in your own PR.

File: Mathlib/Data/NNRat/Defs.lean [hunk 5]
```diff
@@ -367,6 +367,7 @@ theorem natAbs_num_coe : (q : ℚ).num.natAbs = q.num := rfl
 
 lemma coprime_num_den (q : ℚ≥0) : q.num.Coprime q.den := by simpa [num, den] using Rat.reduced _
 
+-- TODO: Rename `Rat.coe_nat_num`, `Rat.intCast_den`, `Rat.ofNat_num`, `Rat.ofNat_den` to match
```
Comment (YaelDillies) [inline Mathlib/Data/NNRat/Defs.lean:370]: Can you take the updated version I gave instead? `Rat.coe_nat_num` has already been renamed.

**Comment by Ruben-VandeVelde** [comment]
Back on the queue, then.

maintainer merge

**Pull request closed without merging**